subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt

```diff
@@ -30,18 +31,26 @@ class DeviceWorker(
         Executors.newSingleThreadExecutor().asCoroutineDispatcher()
     )
 
-    private var state: State = device.state()
-
     fun run() = scope.launch {
 
+        var state: State = try {
+            checkDeviceAlive()
+            device.state()
+        } catch (t: Exception) {
+            device.notifyError("Error while checking device initial state", t)
+            // No intention is lost. DeviceWorkerMessage.WorkerFailed event is unnecessary.
+            // Can't use this device any more. ReservationClient will get a new one.
+            return@launch
+        }
```

Comment:
As I see If we return there we will lose the deviceWorker

Reply:
Yes, this is correct behavior. We can't retry infinitely on the same device.

Killing and recovering device is a responsibility of ReservationClient. We will try to do it by k8s (readiness/liveness probe) in another issue.